src/ExtractorFactory.php

```diff
@@ -46,7 +46,14 @@ public function createExtractor(UriInterface $uri, RequestInterface $request, Re
         $class = $this->default;
 
         foreach ($this->adapters as $adapterHost => $adapter) {
-            if (substr($host, -strlen($adapterHost)) === $adapterHost) {
+            // Check if $host is the same domain as $adapterHost.
+            if ($host === $adapterHost) {
+                $class = $adapter;
+                break;
+            }
+
+            // Check if $host is a subdomain of $adapterHost.
+            if (preg_match('/^([a-z0-9-]+)\.' . preg_quote($adapterHost, '/') . '$/i', $host, $matches)) {
```

Comment:
I'd avoid the use of regex for this.
It would be enough with something like this:

```php
if (substr($host, -strlen($adapterHost + 1)) === ".{$adapterHost}") {

```

Reply:
Changed